test-framework/common/src/main/java/io/quarkus/test/common/NativeImageLauncher.java

```diff
@@ -52,6 +52,7 @@ public void start() throws IOException {
         List<String> args = new ArrayList<>();
         args.add(path);
         args.add("-Dquarkus.http.port=" + port);
+        args.add("-Dquarkus.profile=test");
```

Comment:
Better use `quarkus-profile` for now and get this one in if we can. Then we will change all the occurrences at once.

I tried to do exactly that a few days ago as I was working on a patch needing that and I had weird failures so I would be curious to see what CI has to say.